src/NativeScript/GlobalObject.mm

```diff
@@ -41,11 +42,31 @@
 #include "JSWeakRefPrototype.h"
 #include "JSWeakRefInstance.h"
 #include "TypeFactory.h"
+#include "AllocatedPlaceholder.h"
+#include "ObjCTypes.h"
 
 namespace NativeScript {
 using namespace JSC;
 using namespace Metadata;
 
+JSC::EncodedJSValue JSC_HOST_CALL NSObjectAlloc(JSC::ExecState* execState) {
+    ObjCConstructorBase* constructor = jsCast<ObjCConstructorBase*>(execState->thisValue().asCell());
+    Class klass = constructor->klass();
+    id instance = [klass alloc];
+    GlobalObject* globalObject = jsCast<GlobalObject*>(execState->lexicalGlobalObject());
+
+    if (ObjCConstructorDerived* constructorDerived = jsDynamicCast<ObjCConstructorDerived*>(constructor)) {
+        [instance release];
+        JSValue jsValue = toValue(execState, instance, ^{ return constructorDerived->instancesStructure(); });
+        return JSValue::encode(jsValue);
+    } else if (jsDynamicCast<ObjCConstructorNative*>(constructor) != nullptr) {
+        AllocatedPlaceholder* allocatedPlaceholder = AllocatedPlaceholder::create(execState->vm(), jsCast<GlobalObject*>(execState->lexicalGlobalObject()), AllocatedPlaceholder::createStructure(execState->vm(), globalObject, constructor->instancesStructure()->storedPrototype()), instance, constructor->instancesStructure());
```

Comment:
Rather than creating a new instance of `AllocatedPlaceholder`'s `JSC::Structure` each time, let's keep one instance of the structure per class in `ObjCConstructorNative` and reuse that, just like we do for `ObjCWrapperObject`'s structure.